Lib/pathlib.py

```diff
@@ -463,6 +391,8 @@ def group(self, path):
 
     getcwd = os.getcwd
 
+    realpath = staticmethod(os.path.realpath)
```

Comment:
`os.pah.realpath()` is always non-strict. That's why the implementation is duplicated in pathlib. To use the `os.path` implementation, it needs a `strict=False` keyword-only option.

Reply:
Is checking the return value not sufficient? Trying to avoid touching `os.path` stuff if I can.

Reply:
Maybe that suffices. It could be a symlink to a path that doesn't exist, so you need to check `stat()` to ensure that it exists. That's needed even in non-strict mode, because pathlib always raises for a symlink loop. In POSIX, a symlink loop should fail with `errno` `ELOOP`. I suppose all other errors should be ignored in non-strict mode.

In Windows, a name-surrogate loop (symlink, mountpoint, etc) will fail with `winerror` `ERROR_CANT_RESOLVE_FILENAME` (1921). This will definitely change the behavior of the current implementation, in terms of errors that get ignored and paths that can be partially resolved, since `ntpath.realpath()` is much more fully developed.

Reply:
Really good info, thanks.

What do you think the best course of action would be? I'm happy to add a 'strict mode' to `realpath()` if that looks like the best way forward. There's two things that can be controlled:

- Whether an exception is raised if we encounter a symlink loop (`raise_on_loop=False`?)
- Whether we expect the entire path to exist (`raise_on_missing=False`?)

Reply:
In fact, maybe those two are essentially the same thing. The question is whether we proceed with a "best effort" resolution of the path by tacking on the remaining path components, or give up and raise an exception if we hit an error (not found, symlink loop, etc). That can fit under a single `strict` keyword-only argument to `realpath`. Updating patch incoming...

Reply:
OK, new patch for your consideration. This one adjusts `realpath()` as follows:

- Always raise an `OSError(errno.ELOOP, ...)` when a symlink loop is detected, rather than returning a path that still includes symlinks. Rationale why this behaviour change is OK:
  - The previous behaviour makes no sense and just kicks the symlink-loop issue down the road
  - POSIX `realpath` is documented to return ELOOP when it encounters a symlink loop
  - Surely no one can be relying on this!
- Add a keyword-only `strict` argument that raises an `OSError` if the path (or an ancestor) is inaccessible/missing when `True`.